### ISSUE TITLE
XML injection on search event

### ISSUE THREAD

**Issue by mischabraam** [squeezely/magento2-plugin#31]
With the current module version, it's possible to inject XML using search parameters in the query. This is a security issue. 
When performing a search task like with the following URL the security issue becomes visible. 

https://www.yourshop.com/nl-nl/catalogsearch/result/index/?q=%22%27%3E%3Cqss%20a%3DX160246500Y1_2Z%3E

This results in....

```
<script type="text/javascript">
window._sqzl = _sqzl || []; _sqzl.push({"event":"Search","keyword":"\"'><qss a=X160246500Y1_2Z>"})
</script>
```

In the class `\Squeezely\Plugin\Service\Api\DataLayer` you should escape the data. See attached patch for a quick fix.

[SQUEEZELY_Escape_xss_in_url.composer.patch.zip](https://github.com/squeezely/magento2-plugin/files/10104959/SQUEEZELY_Escape_xss_in_url.composer.patch.zip)

**Comment by Marvin-Magmodules** [comment]
Hi @mischabraam, thank you for opening this issue and adding the patch. We have added this hotfix to release v2.8.1 and have added a separate internal task to review this part for the next release.

**Comment by mischabraam** [comment]
I've introduced a bug in my patch, my bad 😥. 
The function `getSafeData` blindly escapes the value. But it can be an array, apparently. 

```
    protected function getSafeData($object)
    {
        $data = $this->jsonSerializer->unserialize($this->jsonSerializer->serialize($object));
        foreach ($data as $key => $val) {
            if (!is_array($val)) {
                $data[$key] = $this->escaper->escapeXssInUrl($val);
            }
        }

        return $data;
    }
```

This is better.

**Comment by Marvin-Magmodules** [comment]
Hi @mischabraam, yes this was not correctly working with ViewContent event. We’ve just applies this patch and released v2.8.2. We are closing this issue now but please feel free to reopen if you still encounter any issues. Thanks for your help!
